File: user-service/src/main/java/com/nowakArtur97/myMoments/userService/domain/resource/UserModel.java
package com.nowakArtur97.myMoments.userService.domain.resource;

import com.nowakArtur97.myMoments.userService.domain.authentication.AuthenticationResponse;
import com.nowakArtur97.myMoments.userService.domain.common.User;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.*;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@ApiModel(description = "Details about the User")
public class UserModel implements User {

    @ApiModelProperty(notes = "The user's name")
    private String username;

    @ApiModelProperty(notes = "The user's email")
    private String email;

    @ApiModelProperty(notes = "API key")
    private AuthenticationResponse authenticationResponse;

    @ApiModelProperty(notes = "The user's profile")
    private UserProfileModel profile;

    @ApiModelProperty(notes = "The user's roles")
    private List<RoleModel> roles;
}

File: client/src/app/auth/models/user.model.ts
interface AuthenticationResponse {
  readonly token: string;
  readonly expirationTimeInMilliseconds: number;
}

interface UserProfile {
  readonly about: string;
  readonly gender: string;
  readonly interests: string;
  readonly languages: string;
  readonly location: string;
  readonly image: string[];
}

interface Role {
  readonly name: string;
}

export default interface User {
  readonly username: string;
  readonly email: string;
  readonly authenticationResponse: AuthenticationResponse;
  readonly profile: UserProfile;
  readonly roles: Role[];
}

export { AuthenticationResponse, UserProfile, Role, User };
